libs/designer/src/lib/ui/CustomNodes/ScopeCardNode.tsx

```diff
@@ -334,7 +334,10 @@ const ScopeCardNode = ({ data, targetPosition = Position.Top, sourcePosition = P
             nodeIndex={nodeIndex}
           />
           {showCopyCallout ? <CopyTooltip targetRef={rootRef} hideTooltip={clearCopyCallout} /> : null}
-          {isMonitoringView && normalizedType === constants.NODE.TYPE.FOREACH ? (
+          {isMonitoringView &&
```

Comment:
How come adding more checks is making this true to viewing the loopspager but with fewer checks dit was returning null?

Reply:
This bug item goes over more details: 
   - https://msazure.visualstudio.com/One/_workitems/edit/28724205/
   
I discussed this with @ccastrotrejo offline, but basically what we believe is happening is that if a call fails for the repetitions or scopeRepetitions API, the data required to determine the iterations/values won't be present when the LoopsPager attempts to render. 

When this happens the buttons for navigating for iterations won't appear for the nested actions, which results in the bug above. In both PPUX and LAUX, subsequent calls can be made to the same APIs which will give the correct data, but the UI doesn't refresh with the new API calls. 

Based on what we observed, calls to the repetitions APIs for active flow runs that fail typically do so when done for actions that haven't started or are in progress (e.g. the inner for each actions). So, what I noticed is that if we waited to render until after the state was no-longer InProgress, we would have the proper information and could render the buttons as expected.

Reply:
@preetriti1 Is there a specific concern you have with this change?

Reply:
Hey @AfroJoe759 I think this one might not be the approach to solve this. As Priti said, adding more checks to the decision doesn't actually renders the pager. Specially because these are `&&` checks and not `||` checks.

Reply:
So whatever you mentioned, if this is in Progress you do not want to render the pager if this is what we want, @ccastrotrejo then lets check the data if is in progress and put it in useMemo or useEffect block so whenever status changes UI is re-rendered, you can do it either in the pager itself by passing the data as props or from state or do it here.

If something is in effect of some data is being retrieved then we should use the react hooks for re-render and it makes it reliable

Reply:
@preetriti1 @ccastrotrejo sounds good, if you both are fine with the current approach I can move the logic to a useMemo or useEffect block instead.

Reply:
Ok yes. So to be extra clear about this.
What @AfroJoe759 expects to happen over here is that once outer loops completes, the inner loop pages most render.
So please do as Priti said. Figure out a way to render catch the update of it, this can be in a useEffect or useMemo, depending on the implementation